File: sdk/skillbattle/api/ai.py
class AIAPI:

    def __init__(

        self,

        client,

    ):

        self.client = client

    def code_review(

        self,

        payload,

    ):

        return self.client.post(

            "/code-review/review",

            json=payload,

        ).json()

    def battle_coach(

        self,

        payload,

    ):

        return self.client.post(

            "/battle-coach/analyze",

            json=payload,

        ).json()

    def learning_plan(

        self,

        payload,

    ):

        return self.client.post(

            "/learning-engine/generate",

            json=payload,

        ).json()

File: frontend/sdk/js-sdk/src/api/ai.ts
import { AxiosInstance } from "axios";

export class AIAPI {

    constructor(
        private client: AxiosInstance,
    ) {}

    async codeReview(payload: any) {

        return (

            await this.client.post(

                "/code-review/review",

                payload,

            )

        ).data;

    }

    async battleCoach(payload: any) {

        return (

            await this.client.post(

                "/battle-coach/analyze",

                payload,

            )

        ).data;

    }

    async learningPlan(payload: any) {

        return (

            await this.client.post(

                "/learning-engine/generate",

                payload,

            )

        ).data;

    }

}